tests/test_client_regress/tests.py

```diff
@@ -508,6 +508,15 @@ def test_redirect_scheme(self):
             with self.assertRaises(AssertionError):
                 self.assertRedirects(response, 'http://testserver/secure_view/', status_code=302)
 
+    def test_redirect_fetch_redirect_response(self):
+        "if original request follow=false and fetch_redirect_response=True then test client will preserver HTTP headers."
+        response = self.client.get('/redirect_based_on_headers_1/', follow=False, HTTP_REDIRECT="val")
```

Comment:
The test should probably use `subTest()` similar to 272f685794de0b8dead220ee57b30e65c9aa097c so that all HTTP methods are tested (at least if the change to every method is required).